### PR TITLE
[DE] Add `Contribution_guide` and all subarticles

### PR DESCRIPTION
**Pull request by The-Last-Cookie**
The following PR aims to add all articles of the `contribution guide`.

At first, I wanted to add them separately to make the review easier, but since they all cross-link to each other, it's best to have them in one PR. This may make the link checking in the CI checks not going nuts on me as well as simplify the merge process.

## Terminology

As described in #7956, the articles that mention git and git-related workflows use a certain terminology which is explained here again for convenience:

- `pull request` --> `Änderungsvorschlag`, `Änderungsanfrage` and `Änderungsantrag`
- `repositories` --> `Repositorys`
- `merge` --> `zusammenführen` and `Anfrage anwenden/akzeptieren`
- `fork` --> `Verzweigung`
- `branch` --> `Branch` (*perhaps* `Ast` (in any case this should not be too similar to `Abzweigung`))
- `commit` (`to commit`) - `Commit` (`in die Versionshistorie aufnehmen`)
- `contributor` --> `Mitwirkender` and `Beteiligter`

Additionally, these were the terms that I came across while translating:

- `cloning` --> `Klonen`
- `branching` --> `Branching`

## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)
- [x] *(translations only)* The changes are reviewed on GitHub [by a fluent speaker](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#review)

### CODE REVIEW

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+## Einführung
+
+*Für mehr Informationen über git und GitHub, siehe [GitHub-Dokumentation (auf Englisch)](https://docs.github.com)*
+
+**Git** ist ein System zur Versionsverwaltung, das dabei hilft, Änderungen an Dateien zu verwalten. Die Daten und Änderungsgeschichte des osu!-Wikis werden in einem Git-Repository gespeichert. **GitHub** ist eine Entwicklungsplattform, die eine Webschnittstelle für Git-Repositorys bietet und eine Reihe von Projektmanagement-Tools bereitstellt.
+
+## Die Verzweigung synchronisieren
+
+Um Änderungen an einem Repository auf GitHub vorzunehmen, muss sich ein Mitwirkender eine isolierte Kopie des Repositorys beschaffen, welche *Verzweigung* (Fork) genannt wird. Wenn du deine Verzweigung des Repositorys `osu-wiki` erstellst, fertigst du eine Momentaufnahme des Inhalts zu diesem Augenblick an. **Synchronisiere immer deine Verzweigung** zum Beitragen, bevor du Änderungen vornimmst — das kann direkt in GitHub gemacht werden:
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:13]:
```suggestion
Um Änderungen an einem Repository auf GitHub vorzunehmen, muss sich ein Mitwirkender eine isolierte Kopie des Repositorys beschaffen, welche *Verzweigung* (Fork) genannt wird. Wenn du deine Verzweigung des Repositorys `osu-wiki` erstellst, fertigst du eine Momentaufnahme des Inhalts zu diesem Augenblick an. Um einen sinnhaften Beitrag zu erstellen, **synchronisiere immer deine Verzweigung**, bevor du Änderungen vornimmst — das kann direkt in GitHub gemacht werden:
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+1. Gehe zu deiner Verzweigung des Repositorys `osu-wiki`.
+2. Wähle den Branch `master` aus dem Dropdown-Menü aus.
+3. Klicke auf `Fetch upstream` und wähle `Fetch and merge`.
+
+![](img/update-branch.png "Den nicht mehr aktuellen Branch aktualisieren")
+
+Jetzt ist dein Branch auf dem neusten Stand mit dem ursprünglichen Repository.
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:21]:
```suggestion
Jetzt ist dein Branch auf dem selben Stand wie das ursprüngliche Repository.
```

🤔

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+- [Synchronisiere den Branch `master`](#die-verzweigung-synchronisieren).
+- Starte die Arbeit immer mit dem Erstellen eines neuen Branches, der aus dem Branch `master` hervorgeht, und mache nur dort Änderungen. Vergib einen sinnvollen Namen wie `update-staff-log`.<!-- https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow is the explanation, but it doesn't really fit in here -->
+- Nehme deine Arbeit in die Versionshistorie auf, sobald du vernünftig bemessene Änderungen gemacht hast. Es ist besser, einen Artikel als Ganzes zu übertragen, als 10 kleine Änderungen vorzunehmen.
+- **Verwende kurze und sinnvolle Commit-Nachrichten**, da sie andere wissen lassen, was sie erwartet. So etwas wie `Rewrite the section about jump patterns` sagt viel mehr aus als `Update en.md`.
+
+## Eine Änderungsanfrage erstellen
+
+Eine Änderungsanfrage (Pull-Request) zeigt anderen Personen, wie deine Bearbeitung die Dateien beeinflusst. Füge ein paar Informationen zu deiner Anfrage hinzu, um deine Intention zu erklären:
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:60]:
```suggestion
Eine Änderungsanfrage (Pull-Request) zeigt anderen Personen, wie deine Änderungen die Dateien beeinflussen. Füge ein paar Informationen zu deiner Anfrage hinzu, um deine Intention zu erklären:
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+   1. GitHub hebt Konfliktbereiche hervor. Finde eine davon.
+   2. Alles von `<<<<<<<` bis `=======` sind deine Änderungen, wohingegen alles von `=======` bis `>>>>>>> master` ist, was der Branch `ppy/master` enthält.
+   3. Jetzt musst du den Konflikt manuell beheben und die Zeilen mit den Markierungen `<<<<<<<`, `=======` und `>>>>>>> master` löschen.
+   4. Wiederhole den Prozess für alle Konflikte.
+   5. Klicke nach Abschluss auf `Mark as resolved` (das ist nur verfügbar, wenn alle Konfliktbereiche der Datei beseitigt wurden).
+
+2. Falls der Button `Resolve conflicts` gesperrt ist, weil die Konflikte für GitHub zu kompliziert sind, hast du Pech gehabt und musst [deinen Branch aktualisieren](#die-verzweigung-synchronisieren) und deine Änderungen nochmal machen.
+   - *Anmerkung: Das stimmt nur, wenn du in der Verwendung der GitHub-Webschnittstelle eingeschränkt bist.* Es gibt immer noch Möglichkeiten, das Problem zu lösen, aber sie gehören nicht in den Rahmen dieses Artikels. Außerdem ist es vermutlich den Aufwand nicht wert, da du trotzdem die widersprüchlichen Änderungen überschreiben und rückgängig machen musst.
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:101]:
```suggestion
   - *Anmerkung: Das stimmt nur, wenn du auf die Verwendung der GitHub-Webschnittstelle eingeschränkt bist.* Es gibt immer noch Möglichkeiten, das Problem zu lösen, aber sie gehören nicht in den Rahmen dieses Artikels. Außerdem ist es vermutlich den Aufwand nicht wert, da du trotzdem die widersprüchlichen Änderungen überschreiben und rückgängig machen musst.
```

File: wiki/osu!_wiki/Contribution_guide/GitHub_web-based_editor/de.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+3. [Erstelle einen Commit](#änderungen-in-die-versionshistorie-aufnehmen), sobald du mit dem/den Artikel(n) zufrieden bist.
+
+### Neue Artikel oder Übersetzungen
+
+Artikel werden innerhalb von Ordnern mit dem originalen Text (`en.md`) und seinen Übersetzungen gespeichert, die verschiedene sprachbasierte Dateinamen nutzen.
+
+- Um eine neue **Übersetzung** für einen existierenden Artikel hinzuzufügen, klicke mit der rechten Maustaste in den jeweiligen Ordner und lege eine neue Datei mit der Endung `.md` und [einem der unterstützten Dateinamen](/wiki/Article_styling_criteria/Formatting#locales) an.
+- Mache das Folgende für das Erstellen eines neuen **Artikels**:
+  - Erstelle einen Ordner in einer angemessenen Kategorie, die der [Namenskonvention](/wiki/Article_styling_criteria/Formatting#ordner--und-dateistrukturen) folgt. Wenn der Artikel nirgendwo hin gehört, dann kreiere einen Ordner im Verzeichnis `/wiki/`.
+  - Füge die Datei `de.md` mit dem Text des Artikels zum neuen Ordner hinzu.
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/GitHub_web-based_editor/de.md:49]:
```suggestion
  - Füge die Datei `en.md` mit dem Text des Artikels zum neuen Ordner hinzu.
```

From my understanding, one would never write *new* articles in german first.

File: wiki/osu!_wiki/Contribution_guide/de.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# Beitragsleitfaden des osu!-Wikis
+
+Danke für dein Interesse, das osu!-Wiki besser zu machen! Dieser Artikel beschäftigt sich Schritt-für-Schritt mit dem Beteiligungsprozess. Solltest du mit GitHub vertraut sein, folge gerne dem üblichen Feature-Branch-Workflow, um Änderungen vorzuschlagen und springe zum Abschnitt [Selbstüberprüfung](#selbstüberprüfung).
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/de.md:3]:
```suggestion
Danke für dein Interesse, das osu!-Wiki besser zu machen! Dieser Artikel beschäftigt sich Schritt für Schritt mit dem Beteiligungsprozess. Solltest du mit GitHub vertraut sein, folge gerne dem üblichen Feature-Branch-Workflow, um Änderungen vorzuschlagen und springe zum Abschnitt [Selbstüberprüfung](#selbstüberprüfung).
```

The dashes would only make sense if it was part of one word like `Schritt-für-Schritt-Anleitung`

File: wiki/osu!_wiki/Contribution_guide/de.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+- **Struktur und Formatierung**: Siehe dir deinen Artikel mit einem Tool wie dem [Markdown-Editor von jbt](https://jbt.github.io/markdown-editor/) an, um sicherzugehen, dass er genauso aussieht, wie du es möchtest.
+- **Alle Bilder und andere Dateien ohne Text** müssen [unter 1 Megabyte](/wiki/Article_styling_criteria/Formatting#dateigröße) sein. Bei **Screenshots** müssen der standardmäßige osu!-Skin verwenden werden und [spezielle Einstellungen](/wiki/Article_styling_criteria/Formatting#screenshots-des-spielverlaufs), einschließlich der maximalen Abmessung von 1280 × 720, gesetzt sein.
+
+Für mehr spezifische Schreib- und Formatierungsstandards wird auf die [Artikelgestaltungskriterien](/wiki/Article_styling_criteria) verwiesen. Es wird nicht empfohlen, sie auf einmal durchzulesen — ein Rezensent wird wahrscheinlich darauf hinweisen, wenn etwas korrigiert werden muss.
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/de.md:39]:
```suggestion
Für mehr spezifische Schreib- und Formatierungsstandards wird auf die [Artikelgestaltungskriterien](/wiki/Article_styling_criteria) verwiesen. Es wird nicht empfohlen, sie alle auf einmal durchzulesen — ein Rezensent wird wahrscheinlich darauf hinweisen, wenn etwas korrigiert werden muss.
```

File: wiki/osu!_wiki/Contribution_guide/de.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+2. Klicke auf den Button `Contribute↓` und wähle `Open pull request`.
+3. Fülle die Details laut [Bewährte Methoden § Eine Änderungsanfrage erstellen](/wiki/osu!_wiki/Contribution_guide/Best_practices#eine-änderungsanfrage-erstellen) aus und klicke auf `Create pull request`.
+
+## Überprüfung
+
+Alle Änderungen des osu!-Wikis werden moderiert. Während dieser Phase weisen andere Mitwirkende auf mögliche Fehler und Möglichkeiten, diese zu beheben, hin. Das geschieht entweder über Freiform oder indem sie direkt Bearbeitungen an der Änderungsanfrage vorschlagen. Wie auch mit anderen Begutachtungen musst du dich mit den Kommentaren auseinandersetzen, indem du die [Vorschläge anwendest](/wiki/osu!_wiki/Contribution_guide/Best_practices#vorschläge-anwenden) oder indem du erklärst, warum du die Dinge lieber so lässt, wie sie sind.
+
+Wenn sich nach einigen Tagen niemand gemeldet hat, versuche folgendes:
+
+- Stelle sicher, dass du alle Vorschläge beseitigt hast — es könnten Personen auf deine Antwort warten.
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/de.md:55]:
```suggestion
- Stelle sicher, dass du auf alle Vorschläge eingegangen bist — es könnten Personen auf deine Antwort warten.
```
suggestion

File: wiki/osu!_wiki/Contribution_guide/de.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+- Frage andere Beteiligte im [osu!dev Discord-Server](/wiki/Community/osu!dev_Discord_server) (Kanal `#osu-wiki`) oder in den GitHub-Kommentaren.
+- Schaue dir einige genehmigte Änderungsanfragen für deine Sprache an und kontaktiere ihre Gutachter und/oder Autoren, um Hilfe für die Übersetzung zu bekommen ([beispielhafte GitHub-Abfrage](https://github.com/ppy/osu-wiki/pulls?q=is%3Apr+is%3Amerged+%5BID%5D)).
+- Frage deine Freunde für einen kurzen Blick!
```
Comment (FamousPig) [inline wiki/osu!_wiki/Contribution_guide/de.md:58]:
```suggestion
- Frage deine Freunde nach einem kurzen Blick!
```

**Review by FamousPig — CHANGES_REQUESTED** [review]
Mostly wording suggestions.

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md (hunk 2 above)
Comment (The-Last-Cookie) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:21]: Sounds good, though `dem selben` should be one word I think.

File: wiki/osu!_wiki/Contribution_guide/de.md (hunk 8 above)
Comment (The-Last-Cookie) [reply wiki/osu!_wiki/Contribution_guide/de.md]: Sounds better! 👍

File: wiki/osu!_wiki/Contribution_guide/GitHub_web-based_editor/de.md (hunk 5 above)
Comment (The-Last-Cookie) [reply wiki/osu!_wiki/Contribution_guide/GitHub_web-based_editor/de.md]: [`Guides/Searching_and_Downloading_Beatmaps`](https://osu.ppy.sh/wiki/en/Guides/Searching_and_Downloading_Beatmaps) still waiting for a translation to this day lol

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md (hunk 4 above)
Comment (The-Last-Cookie) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:101]: I think `beschränkt` would sound better together with `auf die` 🤔

Comment (FamousPig) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:101]: Sure.

File: wiki/osu!_wiki/Contribution_guide/Best_practices/de.md (hunk 2 above)
Comment (FamousPig) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/de.md:21]: Fixed.

**Review by FamousPig — APPROVED** [review]

**Review by TPGPL — APPROVED** [review]

**Pull request merged**